pkg/flist/flist.go

```diff
@@ -603,7 +605,12 @@ func (f *flistModule) Unmount(name string) error {
 func (f *flistModule) FlistHash(url string) (string, error) {
 	// first check if the md5 of the flist is available
 	md5URL := url + ".md5"
-	resp, err := http.Get(md5URL)
+
+	cl := &http.Client{
```

Comment:
the http client in Go actually uses http2 which means it can reuse connections. But of course this won't work if we create a client everytime a client is needed.

I suggest that may be we instead share the client, maybe make it an attribute of the flistModule. this way we won't need to recreate everytime.